python/cudf/bindings/reduce.pyx

```diff
@@ -52,10 +54,21 @@ def apply_reduce(reduction, col):
             result = gdf_sum_of_squares(<gdf_column*>c_col,
                                         <void*>out_ptr,
                                         outsz)
-
-    check_gdf_error(result)
+        elif reduction == 'product':
+            result = gdf_product(<gdf_column*>c_col, <void*>out_ptr, outsz)
+        else:
+            # TODO: should be GDF_NOTIMPLEMENTED_ERROR
```

Comment:
Why is this TODO rather than just doing it?

Reply:
Done.